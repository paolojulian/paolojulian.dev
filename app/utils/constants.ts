/**
 * These are the links to the different apps
 */
export const ExternalLinks = {
  articles: process.env.NEXT_PUBLIC_ARTICLES_URL ?? 'http://localhost:3002',
  oldApp: process.env.NEXT_PUBLIC_OLD_APP_URL ?? '',
};

export const Routes = {
  Home: '/',
  About: '/about-me',
  Apps: '/apps',
  Articles: `${ExternalLinks.oldApp}/blogs`,
  Contact: `${ExternalLinks.oldApp}/contact`,
} satisfies Record<string, string>;

export const DynamicRoutes = {
  ArticleItem: (slug: string) => `${ExternalLinks.articles}/blogs/${slug}`,
};
