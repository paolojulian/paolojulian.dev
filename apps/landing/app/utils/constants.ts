/**
 * These are the links to the different apps
 */
export const ExternalLinks = {
  articles: process.env.NEXT_PUBLIC_ARTICLES_URL ?? 'http://localhost:3002',
};

export const Routes = {
  Home: '/',
  About: '/about-me',
  Apps: '/apps',
  Articles: ExternalLinks.articles,
  Contact: '/contact',
} satisfies Record<string, string>;
