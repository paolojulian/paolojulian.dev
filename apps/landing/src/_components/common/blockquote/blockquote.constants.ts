export const DATA_TEST = {
  container: 'blockquote-container'
}