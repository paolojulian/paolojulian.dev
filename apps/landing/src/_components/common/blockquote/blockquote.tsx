import { DATA_TEST } from './blockquote.constants';
import React, { FunctionComponent, HtmlHTMLAttributes } from 'react';

interface Props extends HtmlHTMLAttributes<HTMLQuoteElement> {
  // no props
}

const Blockquote: FunctionComponent<Props> = (props) => {
  return (
    <blockquote
      {...props}
      className='border-l-4 border-slate-300 italic pl-4 text-slate-400 mb-2'
      data-testid={DATA_TEST.container}
    ></blockquote>
  );
};

export default Blockquote;
