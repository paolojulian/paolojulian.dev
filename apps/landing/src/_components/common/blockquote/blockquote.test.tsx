import React from 'react';
import { render, screen } from '@testing-library/react';
import Blockquote from './blockquote';
import { DATA_TEST } from './blockquote.constants';

describe('TESTING Blockquote Component', () => {
  describe('GIVEN a Blockquote component with children', () => {
    const children = <div>This is a blockquote</div>;

    function renderComponent() {
      return render(<Blockquote>{children}</Blockquote>);
    }

    describe('WHEN the Blockquote Component is rendered for snapshot testing', () => {
      it('THEN it should match the snapshot', () => {
        const { asFragment } = renderComponent();
        expect(asFragment()).toMatchSnapshot();
      });
    });

    describe('WHEN the Blockquote Component is rendered', () => {
      beforeEach(() => {
        renderComponent();
      });

      it('THEN it should render the component', () => {
        const blockquote = screen.getByTestId(DATA_TEST.container);
        expect(blockquote).toBeInTheDocument();
      });

      it('THEN it should render as a <blockquote> element', () => {
        const blockquote = screen.getByTestId(DATA_TEST.container);
        expect(blockquote).toBeInTheDocument();
        expect(blockquote).toBeInstanceOf(HTMLQuoteElement);
      });

      it('THEN it should contain the children element', () => {
        const blockquote = screen.getByTestId(DATA_TEST.container);
        const childElement = screen.getByText('This is a blockquote');
        expect(blockquote).toContainElement(childElement);
      });
    });
  });
});
