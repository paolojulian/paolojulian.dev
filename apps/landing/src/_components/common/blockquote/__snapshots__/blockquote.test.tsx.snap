// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TESTING Blockquote Component GIVEN a Blockquote component with children WHEN the Blockquote Component is rendered for snapshot testing THEN it should match the snapshot 1`] = `
<DocumentFragment>
  <blockquote
    class="border-l-4 border-slate-300 italic pl-4 text-slate-400 mb-2"
    data-testid="blockquote-container"
  >
    <div>
      This is a blockquote
    </div>
  </blockquote>
</DocumentFragment>
`;
